csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -359,6 +342,8 @@ await WriteMessageAsync(Flatbuf.MessageHeader.Schema, schemaOffset, 0, cancellat
 
             await BaseStream.WriteAsync(messageData, cancellationToken).ConfigureAwait(false);
             await WritePaddingAsync(messagePaddingLength).ConfigureAwait(false);
+
+            return 4 + messageData.Length + messagePaddingLength;
```

Comment:
Some other places are doing checked operations; do we want/need to do so everywhere?

Reply:
Yes, here is a good place too. I'll add `checked` here.